### DISCUSSION TITLE
Need to import a Native Library into Titanium

### DISCUSSION THREAD

**Discussion by FizixTiramu** [appcelerator/titanium_mobile#14015] [Q&A] [answered]
I suspect this can't be done, but I'd like confirmation for a project.

The client requires an API implementation for connecting to a brand of IoT devices, the IoT device and platform manufaturer has a closed source library which is built against the Android and iOS Native SDK's. Is there any way we could import this into a Titanium project?

**Comment by m1ga** [comment, marked as answer]
You can create new Android/iOS modules for Titanium that use external modules:
https://titaniumsdk.com/guide/Titanium_SDK/Titanium_SDK_How-tos/Extending_Titanium_Mobile/Android_Module_Development_Guide/Android_Module_Quick_Start.html

It just depends on if and how they provide access to their module. Many Android modules can be included in gradle like this:
```
dependencies {
  implementation 'com.airbnb.android:lottie:$lottieVersion'
}
```

and then you can use their code in your custom Titanium module. You expose methods, properties, events,... and then you can use it in Titanium. Here is another tutorial: https://www.fromzerotoapp.com/create-android-modules-for-titanium/

But that requires you to have Java/Kotlin and ObjectivC/Swift knowledge to build those modules.
